spec/lib/api/api_config_spec.rb

```diff
@@ -38,6 +43,13 @@
         dangling = api_feature_identifiers - miq_product_features
         expect(dangling).to be_empty
       end
+
+      def each_product_feature(feature = YAML.load_file("#{MiqProductFeature::FIXTURE_PATH}.yml"), &block)
+        block.call(feature)
+        Array(feature[:children]).each do |child|
+          each_product_feature(child, &block)
+        end
+      end
```

Comment:
👍  on the perf gain.  Though one concern is API having knowledge of the above FIXTURE_PATH. I wonder if the each_product_feature method or some variant of it should live in app/models/miq_product_feature.rb and we'd leverage from the let(:miq_product_features) statement as we do above.

Reply:
I like the idea, although I dislike the idea of increasing the surface of `MiqProductFeature`'s interface with another method - especially one that should never be used elsewhere - it's exposing the model's underlying fixture data. I think we may have to acknowledge that this is not a conventional test (it's checking configuration, not exercising behavior) and may have to do break the rules a bit to provide some quick verification that things aren't grossly misconfigured.

Reply:
We can keep as is for now, especially that this spec is the only user of this new method.  We just have to realize that possible future changes to MiqProductFeature's internals might break this test here, which we'd need to fix separately.